org.eclipse.xtext.util/src/org/eclipse/xtext/util/XtextVersion.xtend

```diff
@@ -72,7 +72,13 @@ class XtextVersion {
 			val url = new URL(Plugin.INSTANCE.baseURL + 'META-INF/MANIFEST.MF')
 			is = url.openStream()
 			val manifest = new Manifest(is)
-			return manifest.getMainAttributes().getValue('Maven-Version')
+			val bundleVersion = manifest.getMainAttributes().getValue('Bundle-Version')
```

Comment:
i am not sure what the reasoning behind this was. and this wont work for nightlys. their bundle version looks like `/Users/dietrich/Downloads/EclipseOxygenXtext214Snapshots.app/Contents/Eclipse/plugins/org.eclipse.xtext_2.14.0.v20180126-1316.jar`

so maybe you can introduce looking at BundleVersion as fallback if Maven version returns unspecified

Reply:
same for release candidates/milestones